purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/OfflineEntitlementsIntegrationTest.kt

```diff
@@ -0,0 +1,211 @@
+package com.revenuecat.purchases
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.common.networking.RCHTTPStatusCodes
+import com.revenuecat.purchases.common.sha1
+import com.revenuecat.purchases.factories.StoreProductFactory
+import com.revenuecat.purchases.factories.StoreTransactionFactory
+import com.revenuecat.purchases.helpers.mockQueryProductDetails
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import io.mockk.every
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class OfflineEntitlementsIntegrationTest : BasePurchasesIntegrationTest() {
+
+    private lateinit var server: MockWebServer
+
+    private val testProductId = "annual_freetrial"
+    private val testBasePlanId = "p1y"
+    private val productEntitlementMappingMockResponse = """
+        {
+          "product_entitlement_mapping": {
+            "$testProductId": {
+              "base_plan_id": "$testBasePlanId",
+              "entitlements": [
+                "pro_cat"
+              ],
+              "product_identifier": "$testProductId"
+            }
+          }
+        }
+    """.trimIndent()
+    private val customerInfoCacheString = """
+        {
+          "request_date": "2023-05-16T14:06:28Z",
+          "request_date_ms": 1684245988765,
+          "subscriber": {
+            "entitlements": {},
+            "first_seen": "2023-05-16T14:06:28Z",
+            "last_seen": "2023-05-16T14:06:28Z",
+            "management_url": null,
+            "non_subscriptions": {},
+            "original_app_user_id": "$testUserId",
+            "original_application_version": null,
+            "original_purchase_date": null,
+            "other_purchases": {},
+            "subscriptions": {}
+          },
+          "schema_version": 3,
+          "customer_info_request_date": 1684245988766
+        }
+    """.trimIndent()
+    private val cacheKeyPrefix = "com.revenuecat.purchases.${Constants.apiKey}"
+    private val productEntitlementMappingCacheKey = "$cacheKeyPrefix.productEntitlementMapping"
```

Comment:
It's not great to duplicate these cache keys... But I didn't want to expose them in the `common` module just for this... I think it's ok? Lmk what you think